File: internal/pkg/database/models/iam_credentials_sessions.go
package models

import (
	"time"

	"github.com/gofrs/uuid/v5"
)

type IamCredentialsSessions struct {
	OpenidSid []uuid.UUID `json:"openid_sid,omitempty"`
	OpenidSub []uuid.UUID `json:"openid_sub,omitempty"`
	CreatedOn []time.Time `json:"created_on,omitempty"`
	ExpiresOn []time.Time `json:"expires_on,omitempty"`
}

type iamCredentialsSessionsTable struct {
	TableName string

	OpenidSid string
	OpenidSub string
	CreatedOn string
	ExpiresOn string
}

func IamCredentialsSessionsTable() iamCredentialsSessionsTable {
	return iamCredentialsSessionsTable{
		TableName: "iam_credentials_sessions",

		OpenidSid: "openid_sid",
		OpenidSub: "openid_sub",
		CreatedOn: "created_on",
		ExpiresOn: "expires_on",
	}
}

File: web/src/lib/database/iam_credentials_sessions.ts
namespace _IamCredentialsSessions {
	export enum Column {
		OpenidSid = 'openid_sid',
		OpenidSub = 'openid_sub',
		CreatedOn = 'created_on',
		ExpiresOn = 'expires_on'
	}

	export const TableName = 'iam_credentials_sessions'

	export interface Interface {
		openid_sid?: string[]
		openid_sub?: string[]
		created_on?: string[]
		expires_on?: string[]
	}
}

export default _IamCredentialsSessions
